homeassistant/components/devolo_home_control/siren.py

```diff
@@ -0,0 +1,86 @@
+"""Platform for siren integration."""
+from typing import Any
+
+from devolo_home_control_api.devices.zwave import Zwave
+from devolo_home_control_api.homecontrol import HomeControl
+
+from homeassistant.components.siren import (
+    ATTR_TONE,
+    SUPPORT_TONES,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SirenEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .devolo_multi_level_switch import DevoloMultiLevelSwitchDeviceEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Get all binary sensor and multi level sensor devices and setup them via config entry."""
+    entities = []
+
+    for gateway in hass.data[DOMAIN][entry.entry_id]["gateways"]:
+        for device in gateway.multi_level_switch_devices:
+            for multi_level_switch in device.multi_level_switch_property:
+                if multi_level_switch.startswith("devolo.SirenMultiLevelSwitch"):
+                    entities.append(
+                        DevoloSirenDeviceEntity(
+                            homecontrol=gateway,
+                            device_instance=device,
+                            element_uid=multi_level_switch,
+                        )
+                    )
+
+    async_add_entities(entities, False)
+
+
+class DevoloSirenDeviceEntity(DevoloMultiLevelSwitchDeviceEntity, SirenEntity):
+    """Representation of a cover device within devolo Home Control."""
+
+    def __init__(
+        self, homecontrol: HomeControl, device_instance: Zwave, element_uid: str
+    ) -> None:
+        """Initialize a devolo multi level switch."""
+        super().__init__(
+            homecontrol=homecontrol,
+            device_instance=device_instance,
+            element_uid=element_uid,
+        )
+        self._attr_available_tones = [
+            *range(
+                self._multi_level_switch_property.min,
+                self._multi_level_switch_property.max + 1,
+            )
+        ]
+        self._attr_supported_features = (
+            SUPPORT_TURN_OFF | SUPPORT_TURN_ON | SUPPORT_TONES
```

Comment:
I noticed some of these were not listed in the documentation for `siren` https://developers.home-assistant.io/docs/core/entity/siren -- consider updating that too.